src/main/java/com/github/monster860/fastdmm/FastDMM.java

```diff
@@ -262,559 +271,654 @@ public void initSwing() {
+                                RenderInstance ri = new RenderInstance(currCreationIndex++);
+                                ri.plane = 101;
+                                ri.x = x;
+                                ri.y = y;
+                                ri.substate = interface_dmi.getIconState("" + dirs).getSubstate(2);
+                                ri.color = new Color(200,200,200);
+
+                                rendInstanceSet.add(ri);
+                            }
+                        }
+                    }
+                }
+            }
+
+
+            if(currPlacementHandler != null) {
+                currCreationIndex = currPlacementHandler.visualize(rendInstanceSet, currCreationIndex);
+            }
+
+            currCreationIndex = placementMode.visualize(rendInstanceSet, currCreationIndex);
+
+            for(RenderInstance ri : rendInstanceSet) {
+                glColor3f(ri.color.getRed()/255f, ri.color.getGreen()/255f, ri.color.getBlue()/255f);
+                glBindTexture(GL_TEXTURE_2D, ri.substate.dmi.glID);
+
+                glPushMatrix();
+                glTranslatef(ri.x, ri.y, 0);
+                glBegin(GL_QUADS);
+                glTexCoord2f(ri.substate.x2, ri.substate.y1);
+                glVertex3f( -.5f + (ri.substate.dmi.width/(float)objTree.icon_size), -.5f + (ri.substate.dmi.height/(float)objTree.icon_size), 0);
+                glTexCoord2f(ri.substate.x1, ri.substate.y1);
+                glVertex3f(-.5f, -.5f + (ri.substate.dmi.height/(float)objTree.icon_size), 0);
+                glTexCoord2f(ri.substate.x1, ri.substate.y2);
+                glVertex3f(-.5f,-.5f, 0);
+                glTexCoord2f(ri.substate.x2, ri.substate.y2);
+                glVertex3f( -.5f + (ri.substate.dmi.width/(float)objTree.icon_size),-.5f, 0);
+                glEnd();
+                glPopMatrix();
+            }
+
+            glBindTexture(GL_TEXTURE_2D, -1);
+            glColor4f(1, 1, 1, .25f);
+            glPushMatrix();
+            glTranslatef(selX, selY, 1);
+            glBegin(GL_QUADS);
+            glVertex3f( .5f, .5f, 0);
+            glVertex3f(-.5f, .5f, 0);
+            glVertex3f(-.5f,-.5f, 0);
+            glVertex3f( .5f,-.5f, 0);
+            glEnd();
+            glPopMatrix();
+
+            Display.update();
+
+            processInput();
+
+            Display.sync(60);
+        }
+    }
+
+    private String getAppData() {
+        String path = System.getProperty("user.home");
+        if(path.contains("/")) { //we need this due to differences between OSs
+            path += "/.fastdmm/";
+        } else {
+            path += "\\.fastdmm\\";
+        }
+        Path AppDataPath = Paths.get(path);
+        if(Files.notExists(AppDataPath)) {
+            new File(path).mkdirs();
+        }
+
+        return path;
+    }
+
+    private void initRecent(JMenu menutoadd) {
+        menutoadd.removeAll();
+        String recentFile = getAppData() + "recent.txt";
```

Comment:
`getAppData()` can be replaced with `System.getProperty("user.home") + File.separator + ".fastdmm"` really.